### PR TITLE
Update AWS HA autoscale and starter cluster examples to use hardened AMIs with Graviton/ARM64

### LINKED ISSUE #2734: Modify reference Terraform to use Prometheus instead of Telegraf
## Description

There are several issues with grafana

~### Set default value of grafana pass in example to something random~

~Many users will end up with grafana's opened up with "setup some password here" as a result of this:~

~https://github.com/gravitational/teleport/blob/689bb5b4f8681e836eb73b881e1b59533cfe2462/examples/aws/terraform/Makefile#L37~

~Please set it to `hexdump -n 16 -e '4/4 "%08X" 1 "\n"' /dev/urandom` by default, so every user will get somewhat random default~

~### Default grafana instance that is spun up has no datasource, or default dashboard configured, what is a regression.~

~### Update `curl` commands to use new syntax~

~Instead of supplying password within http url, use `curl -u "admin:${GRAFANA_PASS}" syntax~

~### Auth server has empty target URL~

```
[[outputs.influxdb]]
  ## The full HTTP or UDP endpoint URL for your InfluxDB instance.                                                                                                                                                                                                                        
  ## Multiple urls can be specified as part of the same cluster,                                                                                                                                                                                                                          
  ## this means that only ONE of the urls will be written to each interval.                                                                                                                                                                                                               
  urls = [""] # required                                                                                                                                                                                                                                                                  
  ## The target database for metrics (telegraf will create it if not exists).                                                                                                                                                                                                             
  database = "telegraf" # required  
```

### Switch to prometheus

And the last, but not least :) Let's switch to prometheus as a backing data store, and remove telegrafs as well, probably this will render some of the issues above obsolete, but worth doing as it will simplify things.

### LINKED ISSUE #4572: Grafana dashboard supplied with HA reference Terraform has blank tables/entities
**What happened**: The Grafana dashboard for "Teleport Health Stats" that we supply with our HA reference Terraform has some blank entries in it - looks like the queries either aren't valid for the specified metrics any more or there's been a breaking change to the InfluxDB queries.

2020-10-16-135951__1625x1369__maim

**What you expected to happen**: All items on the reference "Teleport Health Stats" dashboard should contain useful data.

The dashboard config is here: https://github.com/gravitational/teleport/blob/master/examples/aws/terraform/ha-autoscale-cluster/assets/health-dashboard.json

As per https://github.com/gravitational/teleport/issues/2734, we should also migrate to using Prometheus directly rather than Telegraf.

**How to reproduce it (as minimally and precisely as possible)**: Set up a Terraform using https://github.com/gravitational/teleport/tree/master/examples/aws/terraform/ha-autoscale-cluster, load the Grafana dashboard (https://<url>:8443), log in with `admin` and the configured password (`TF_VAR_grafana_pass`) and look at the dashboard.

### Environment
- Teleport version (use `teleport version`): `Teleport Enterprise v4.4.0-rc.3 git:v4.4.0-rc.3-0-g32997e9ac go1.14.4`

### LINKED ISSUE #4756: Replace nginx in aws terraform with Teleport application access
Right now our AWS terraform sets grafana with cert and nginx and a custom password.

With 5.0 we can replace all this with connecting grafana with default username/password with teleport application access.

### PR DESCRIPTION
**Pull request by reedloden**
AWS starter and HA autoscale cluster examples are outdated now that legacy AMIs have been deprecated. Update them to use hardened AMIs and bump AWS terraform provider to v5.x.

Graviton is the preferred (and cheaper) platform on EC2, so our documentation/example code should use such instance types. This also means using ARM64 builds as well.

Remove the `monitor` role and associated infrastructure (as the new hardened AMIs lack telegraf, influxdb, etc.).

s/letsencrypt/Let's Encrypt/

Ref #34282.
Ref #36110.
Closes #2734.
Closes #4756.
Closes #4572.

### CODE REVIEW

**Review by ptgott — APPROVED** [review]

**Comment by ptgott** [comment]
Are we backporting this one?

**Review by webvictim — CHANGES_REQUESTED** [review]
Hold up. The ha-autoscale-cluster deployment example still tells everyone to use our Teleport-published AMIs, which are only built for `amd64` today.

See the instructions here on how to get the right AMI ID for a given region: https://github.com/gravitational/teleport/blob/master/docs/pages/deploy-a-cluster/deployments/aws-ha-autoscale-cluster-terraform.mdx#L176-L189

The published AMIs are built by this script: https://github.com/gravitational/teleport/blob/master/assets/aws/files/install.sh#L56-L107

If we change this, it will potentially break everything for `amd64` users. To do this properly, we will likely need to:
- make a new version of this install script which creates AMIs with `arm64` binaries instead
- edit the Packer manifest (https://github.com/gravitational/teleport/blob/master/assets/aws/single-ami.json) to tell it to how to build the AMIs under a different name
- edit the Drone/GHA build system to automatically publish those AMIs on release
- update the docs to let people choose their arch

As such, I would recommend not backporting this either. We should ideally start publishing `arm64` AMIs alongside `amd64` with a Teleport major version release and some details in the release notes. We could then potentially look at deprecating `amd64` AMIs within a year or two.

**Comment by hugoShaka** [comment]
I agree with gus and think this must not be backported.

I am also uncomfortable with removing the ability to deploy using regular AMD64 instances. Even if we default to ARM instances, this should be configurable.

> Graviton is the preferred (and cheaper) platform on EC2

This is only true for users paying full price. When you cut deal with AWS, they choose which machine you should prefer and heavily reduce the SKUs for those and/or give you a lot of RIs/CSPs. The deals I saw last year were still favouring AMD-based instances.

**Comment by reedloden** [comment]
Appreciate the review and comments. Definitely was unclear what's used here, especially since Ansible is literally downloading a release and installing it.

Wasn't planning on backporting this, fwiw.

The AMIs being referenced are being removed starting with Teleport 15. See #34282. So, if the docs are referring to that, they need to be updated.

It does look like our hardened AMIs are only AMD64, so I will work on updating those to be built on ARM64 as well.

> I am also uncomfortable with removing the ability to deploy using regular AMD64 instances. Even if we default to ARM instances, this should be configurable.

The instance type is already configurable. We're just defaulting to ARM64 / Graviton.

> > Graviton is the preferred (and cheaper) platform on EC2
> 
> This is only true for users paying full price. When you cut deal with AWS, they choose which machine you should prefer and heavily reduce the SKUs for those and/or give you a lot of RIs/CSPs. The deals I saw last year were still favouring AMD-based instances.

Haven't seen those types of deals before. I don't think it's the norm based on our own interactions with AWS. For instance, we have an EDP/PPA in place with AWS, and it covers almost all AWS services with a specific discount percentage. Any EC2-specific discounts are handled with CSPs/RIs (we use CSPs ourselves, as it just makes more sense to not be tied to instance type and region).

File: examples/aws/terraform/ha-autoscale-cluster/ssm.tf [hunk 1]
```diff
@@ -3,17 +3,8 @@
 // is destroyed, cluster will overwrite them with real values
 
 resource "aws_ssm_parameter" "license" {
-  count     = var.license_path != "" ? 1 : 0
-  name      = "/teleport/${var.cluster_name}/license"
-  type      = "SecureString"
-  value     = file(var.license_path)
-  overwrite = true
```
Comment (webvictim) [inline examples/aws/terraform/ha-autoscale-cluster/ssm.tf:10]: Why was `overwrite` removed here?

File: examples/aws/terraform/starter-cluster/ssm.tf [hunk 2]
```diff
@@ -1,8 +1,7 @@
 // SSM for Teleport Enterprise license storage and retrieval
 resource "aws_ssm_parameter" "license" {
-  count     = var.license_path != "" ? 1 : 0
-  name      = "/teleport/${var.cluster_name}/license"
-  type      = "SecureString"
-  value     = file(var.license_path)
-  overwrite = true
```
Comment (webvictim) [inline examples/aws/terraform/starter-cluster/ssm.tf:7]: Why was `overwrite` removed here?

**Review by webvictim — APPROVED** [review]

File: examples/aws/terraform/ha-autoscale-cluster/ssm.tf (hunk 1 above)
Comment (reedloden) [reply examples/aws/terraform/ha-autoscale-cluster/ssm.tf:10]: Ah, good catch. It is deprecated, but it's not removed until v6.0.0 of the provider. I just misread the deprecation notice.

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ssm_parameter states:

> `overwrite` also makes it possible to overwrite an existing SSM Parameter that's not created by Terraform before. This argument has been deprecated and will be removed in v6.0.0 of the provider. For more information on how this affects the behavior of this resource, see [this issue comment](https://github.com/hashicorp/terraform-provider-aws/issues/25636#issuecomment-1623661159).

The default in v6.0.0 will be `true`, but until then, we do still need this.

File: examples/aws/terraform/starter-cluster/ssm.tf (hunk 2 above)
Comment (reedloden) [reply examples/aws/terraform/starter-cluster/ssm.tf:7]: See above.

File: examples/aws/terraform/ha-autoscale-cluster/ssm.tf (hunk 1 above)
Comment (webvictim) [reply examples/aws/terraform/ha-autoscale-cluster/ssm.tf:10]: Cool, TIL!

**Review by hugoShaka — APPROVED** [review]
LGTM, can you rename the PR and add a changelog entry mentioning the removal of the monitoring stack?

I think this fixes: https://github.com/gravitational/teleport/issues/2734

**Comment by reedloden** [comment]
> LGTM, can you rename the PR and add a changelog entry mentioning the removal of the monitoring stack?

Done.

> I think this fixes: #2734

Yup, and a few others. I updated description to close them as well.

**Review by zmb3 — APPROVED** [review]

**Pull request merged**